packages/vite/src/node/ssr/fetchModule.ts

```diff
@@ -105,15 +108,16 @@ export async function fetchModule(
   }
 
   // module entry should be created by transformRequest
-  mod ??= await environment.moduleGraph.getModuleByUrl(url)
+  const modById = environment.moduleGraph.getModuleById(mod.id!)
```

Comment:
why can't you just reassign `mod` here?

Reply:
This was for typing since `ensureEntryFromUrl` ensures it but `getModuleById` can return `undefined`. It's probably safe to do `!` for most practical purposes. Let's see how it goes https://github.com/vitejs/vite/pull/18361/commits/cfc490161256f66faac5f2e1ca6ad3cc6ffd4390

Reply:
Hmm, that sounds right. I just remove more.